qark/plugins/Dynamically_Loading_Code.py

```diff
@@ -1,73 +1,130 @@
+        return
+
+    # recursive function to find the register receiver function name by traversing down the AST
+    def recursive_register_receiver_function(self, fields, f, res):
+        if type(fields) is m.MethodDeclaration:
+            if str(fields.name) == self.DYNAMIC_BROADCAST_RECEIVER:
+                receivers_list.append(file_name)
+            elif self.DYNAMIC_BROADCAST_RECEIVER in str(fields):
+                if file_name not in receivers_list:
+                    receivers_list.append(file_name)
+        elif type(fields) is list:
+            for tree_object_fields in fields:
+                self.recursive_register_receiver_function(tree_object_fields, f, res)
+        elif hasattr(fields, '_fields'):
+            for values in fields._fields:
+                self.recursive_register_receiver_function(getattr(fields, values), f, res)
+        return
+
     def target(self, queue):
         files = common.java_files
-        global parser, tree, fileName
+        global parser, tree, file_name
         parser = plyj.Parser()
         tree = ''
         res = []
-
-        #List of Broadcast Receiver
-        list_BR = []
         count = 0
         for f in files:
             count += 1
             pub.sendMessage('progress', bar=self.name, percent=round(count * 100 / len(files)))
-            fileName = str(f)
+            file_name = str(f)
             try:
+                # Parse the java file to an AST
                 tree = parser.parse_file(f)
-            except Exception:
-                continue
+            except Exception as e:
+                common.logger.exception(
+                    "Unable to parse the file and generate as AST. Error: " + str(e))
+
             try:
                 for import_decl in tree.import_declarations:
+                    # Check if DexClassLoader is called in the import statement; example import dalvik.system.DexClassLoader
                     if self.DEX_CLASS_LOADER in import_decl.name.value:
-                        if self.CLASS_LOADER in str(tree):
-                            PluginUtil.reportInfo(fileName, self.DexClassLoaderIssueDetails(fileName), res)
+                        for type_decl in tree.type_declarations:
+                            # Check class declaration within the java source code
+                            if type(type_decl) is m.ClassDeclaration:
+                                # Traverse through every field declared in the class
+                                for fields in type_decl.body:
+                                    try:
+                                        self.recursive_classloader_function(fields, f, res)
+                                    except Exception as e:
+                                        common.logger.error(
+                                            "Unable to run class loader plugin " + str(e))
+            except Exception:
+                pass
```

Comment:
I'd at least log something here. Not sure exactly what kind of error would be thrown